### ISSUE TITLE
Bar updates on new lines instead of replacing current.

### ISSUE THREAD

**Issue by sargunster** [fishery/spin#1]
In OSX, the bar style updates on newlines instead of replacing the current line. The non-bar styles work fine.
<img width="697" alt="screen shot 2016-02-18 at 10 13 05 pm" src="https://cloud.githubusercontent.com/assets/1320357/13165621/19bc5566-d68d-11e5-8a1f-a0ec64c10dd8.png">

**Comment by bucaran** [comment]
@sargunster Good catch. That's because we are not calculating the number of columns using `tput cols` correctly. You can increase the width of your terminal window and it should display alright.
